docs/snippets/compare_snippet_output.py

```diff
@@ -64,7 +64,7 @@ def __lt__(self, other: Example) -> bool:
         return self.name < other.name
 
     def __repr__(self) -> str:
-        return f"Example(subdir={self.subdir}, name={self.name})"
+        return f"{self.subdir}/{self.name}"
```

Comment:
nit/psa: in the python canon, `__repr__` should ideally return a string that is valid python and interprets as the original object. So the previous version was more correct (though python nerds would rather have `self.__class__.__name__` instead of duplicating the class name). Your new implementation is a better fit for `__str__`, which is the user facing representation of the underlying data (similar to rust's `Display`). In this case, I don't really care, but it is very nice to have a correct repr for any object that has a chance to be used by users in a notebook/REPL context.

Reply:
Actually, the correct way is to make `Example` a `@dataclass.dataclass` and all this boilerplate will be automagically generated, including `__init__`, `__eq__`, `__lt__` (though not with the current implementation), `__repr__`, and possibly `__hash__` and `__str__`.

Reply:
I think I wanted to override `__str__` instead of `__repr__`.